# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

pkg-config --exists libudev-build && REQUIRES="$REQUIRES libudev-build"

# configure (ac-2.69; am-1.15; lt-2.4.6)
libtoolize -c -f; aclocal; autoheader; autoconf; automake -a -c
./configure --disable-static --enable-examples-build --prefix=

# build
make

# install
install -d $DESTDIR/include/libusb-1.0
install -m 644 libusb/libusb.h $DESTDIR/include/libusb-1.0
install -d $DESTDIR/lib
install libusb/.libs/libusb-1.0.so.0 $DESTDIR/lib
ln -sf libusb-1.0.so.0 $DESTDIR/lib/libusb-1.0.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 libusb-1.0.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/libusb-build.pc << EOF
Name: libusb
Version: 1.0.23
Description: USB library
Requires: $REQUIRES

devel=\\
/include/libusb-1.0/libusb.h \\
/lib/libusb-1.0.so \\
/lib/pkgconfig/libusb-1.0.pc

exec=\\
/lib/libusb-1.0.so.0
EOF

exit

# patch
--- libusb-1.0.23.orig/configure.ac
+++ libusb-1.0.23/configure.ac
@@ -27,6 +27,7 @@
 LTLDFLAGS="-version-info ${lt_current}:${lt_revision}:${lt_age}"
 
 AM_INIT_AUTOMAKE
+AM_MAINTAINER_MODE
 
 AC_CONFIG_SRCDIR([libusb/core.c])
 AC_CONFIG_MACRO_DIR([m4])

# source
https://github.com/libusb/libusb/releases/download/v1.0.23/libusb-1.0.23.tar.bz2
